src/Bootstrap/Carousel.elm

```diff
@@ -0,0 +1,509 @@
+-}
+toSlide : Int -> State -> State
+toSlide n =
+    update (StartTransition (Number n))
+
+
+{-| Calculate the next index based on the current state
+
+This is hardcoded to cycle between the first 3 slides for now
+-}
+nextIndex : TransitionStage Transition -> Int -> Int
+nextIndex state currentIndex =
+    let
+        helper transition =
+            case transition of
+                Next ->
+                    (currentIndex + 1) % 3
+
+                Prev ->
+                    (currentIndex - 1) % 3
+
+                Number m ->
+                    m % 3
+    in
+        case state of
+            Start transition ->
+                helper transition
+
+            Animating transition ->
+                helper transition
+
+            NotAnimating ->
+                currentIndex
+
+
+
+--- View ---
+
+
+{-| Opaque type that defines the view configuration of your carousel
+-}
+type Config msg
+    = Config
+        { id : String
+        , toMsg : Msg -> msg
+        , options : List (CarouselOption msg)
+        , slides : List (Slide.Config msg)
+        , controls : Bool
+        , indicators : Bool
+        }
+
+
+view : State -> Config msg -> Html.Html msg
+view model (Config settings) =
+    let
+        (State tstage { currentIndex, wrap }) =
+            model
+
+        size =
+            List.length settings.slides
+
+        indicatorsHtml =
+            if settings.indicators then
+                indicators settings.id (List.length settings.slides) (nextIndex tstage currentIndex)
+            else
+                text ""
+
+        slidesHtml =
+            div [ class "carousel-inner", attribute "role" "listbox" ]
+                (List.indexedMap (viewSlide model) settings.slides)
+
+        controlsHtml =
+            if settings.controls then
+                if wrap || (currentIndex /= 0 && currentIndex /= size - 1) then
+                    [ controlPrev settings.id, controlNext settings.id ]
+                else if currentIndex == 0 then
+                    [ controlNext settings.id ]
+                else
+                    [ controlPrev settings.id ]
+            else
+                []
+
+        defaultCarouselAttributes =
+            -- catch the transitionend event, to end an ongoing transition
+            [ Attributes.id settings.id
+            , class "carousel slide"
+            , on "transitionend" (Decode.succeed (settings.toMsg EndTransition))
+            , onMouseEnter (settings.toMsg <| SetHover True)
+            , onMouseLeave (settings.toMsg <| SetHover False)
+            ]
+    in
+        div (List.concatMap carouselOptionToAttributes settings.options ++ defaultCarouselAttributes)
+            (Html.map settings.toMsg indicatorsHtml :: slidesHtml :: (List.map (Html.map settings.toMsg) controlsHtml))
+
+
+{-| Sets the correct classes to the current and (potentially) next element.
+-}
+viewSlide : State -> Int -> Slide.Config msg -> Html.Html msg
```

Comment:
Again a very chunky let block. I would find it way more pleasing to the eye (and easier to see the big picture) if this was extracted into smaller functions.